cirq/experiments/qubit_characterizations.py

```diff
@@ -0,0 +1,544 @@
+    circuit.append(ops.measure(qubit, key='z'))
+    sweep = study.Linspace(key='theta', start=0.0, stop=final_angle,
+                           length=num_points)
+    results = sampler.run_sweep(circuit, params=sweep, repetitions=num_shots)
+    half_turns = numpy.linspace(0.0, final_angle, num_points)
+    excited_state_probs = numpy.zeros(num_points)
+    for i in range(num_points):
+        excited_state_probs[i] = numpy.mean(results[i].measurements['z'])
+
+    return RabiResult(half_turns, excited_state_probs)
+
+
+def single_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                         qubit: devices.GridQubit,
+                                         num_cfds_seq: Sequence[int],
+                                         num_circuits: int, num_shots: int,
+                                         use_xy_basis: bool = True) -> RBResult:
+    """
+    Clifford-based randomized benchmarking (RB) of a single qubit.
+
+    A total of num_circuits random circuits are generated, each of which
+    contains a fixed number of single-qubit Clifford gates plus one
+    additional Clifford that inverts the whole sequence and a measurement in
+    the z-basis. Each circuit is repeated num_shots times and the average |0>
+    state population is determined from the measurement outcomes of all of the
+    circuits.
+
+    The above process is done for different numbers of Cliffords specified in
+    num_cfds_seq.
+
+    See Barends et al., Nature 508, 500 for details.
+
+    Args:
+        sampler: The quantum engine or simulator to run the circuits.
+        qubit: The qubit under test.
+        num_cfds_seq: The different numbers of Cliffords in the RB study.
+        num_circuits: The number of random circuits generated for each number of
+            Cliffords.
+        num_shots: The number of repetitions of each circuit.
+        use_xy_basis: Determines if the Clifford gates are built with x and y
+            rotations (True) or x and z rotations (False).
+
+    Returns:
+        A RandomizedBenchmarkingResult object that stores and plots the result.
+    """
+
+    c1_in_xy, c1_in_xz, _, _, _ = _single_qubit_cliffords()
+    c1 = c1_in_xy if use_xy_basis else c1_in_xz
+    cfd_mats = numpy.array([_gate_seq_to_mats(gates) for gates in c1])
+
+    gnd_probs = []
+    for num_cfds in num_cfds_seq:
+        excited_probs_l = []
+        for _ in range(num_circuits):
+            circuit = _random_single_q_clifford(qubit, num_cfds, c1, cfd_mats)
+            circuit.append(ops.measure(qubit, key='z'))
+            results = sampler.run(circuit, repetitions=num_shots)
+            excited_probs_l.append(numpy.mean(results.measurements['z']))
+        gnd_probs.append(1.0 - numpy.mean(excited_probs_l))
+
+    return RBResult(num_cfds_seq, gnd_probs)
+
+
+def two_qubit_randomized_benchmarking(sampler: sim.SimulatesSamples,
+                                      q_0: devices.GridQubit,
```

Comment:
Maybe a combined `qubits`? I don't feel strongly about it, just seems like `q_0` is an odd parameter name to expose to the user.

Reply:
I rename q_0 and q_1 to first_qubit and second_qubit. Hopefully these look more intuitive. Same for the signatures in two qubit tomography.